internal/pubsub/mempubsub/mem.go

```diff
@@ -86,20 +109,28 @@ type subscription struct {
 }
 
 // OpenSubscription creates a new subscription for the given topic.
-// Unacknowledged messages will become available for redelivery after ackDeadline.
-func OpenSubscription(t driver.Topic, ackDeadline time.Duration) driver.Subscription {
-	tt := t.(*topic)
+func (b *Broker) OpenSubscription(ctx context.Context, topicName string, ackDeadline time.Duration) *pubsub.Subscription {
+	b.mu.Lock()
+	t := b.topics[topicName]
+	b.mu.Unlock()
+	if t == nil {
+		panic(fmt.Sprintf("no topic named %s", topicName))
```

Comment:
Don't think this should panic. What would a "real" implementation do? IIUC, it would not know at this time, the `Open` would work fine but then you'd get an error when reading.

Reply:
Done.